docs/source/tutorial/interfaces.rst

```diff
@@ -429,10 +427,10 @@ evaluated and then implicitly bound. Conceptually, we can think of
 
     (!) : m a -> a
 
-Note, however, that it is not really a function, merely syntax! In
+Note, however, that it is not really a function, but merely syntax! In
 practice, a subexpression ``!expr`` will lift ``expr`` as high as
 possible within its current scope, bind it to a fresh name ``x``, and
-replace ``!expr`` with ``x``. Expressions are lifted depth first, left
+replace ``!expr`` with ``x``. Expressions are lifted depth-first, left
 to right. In practice, ``!``-notation allows us to program in a more
```

Comment:
Whilst we are revisiting text, `!`-notation should probably be written as 'bang-notation'. 

Generally speaking, however, there is a larger debate to be had if the phrase 'x'-notation should include the name of the notation or the symbol itself.

Should it be do-notation or `dot`-notation? 

and referencing cases further on:

Should it be bang-notation or `!`-notation?

Should it be range-notation or `[A..B]`-notation?

Reply:
My personal preference would be to do both the first time a symbol-based notation is introduced, and then sticking to "how it would appear as code" for the rest of the document, for example:
"""
Idris provides `!`-notation ("bang"-notation) for [...]
Here, we can see `!`-notation used as `!expr` [...]
"""
(Modulo minor symbol/stylistic/phrasing choices)

This way the docs introduce the symbol, which is what the user will actually be writing in their code, while also immediately providing the way to pronounce and refer to it; a bit like introducing acronyms. Ideally I would also want that full version to be in the title of the page (if it is its own page).